modules/Router.js

```diff
@@ -82,7 +84,10 @@ const Router = React.createClass({
       history,
       createRoutes(routes || children)
     )
+
     const router = createRouterObject(history, transitionManager)
+    router.location = null
+    router.params = null
```

Comment:
We can do it in both then? I think part of the reason @ryanflorence wanted to defer this to 3.0 was so that we could implement it in this way, and get away from some of the requirements around keeping track of a stateful router object.